app/controllers/application_controller.rb

```diff
@@ -1,10 +1,145 @@
 require './config/environment'
-
+require 'pry'
 class ApplicationController < Sinatra::Base
+    configure do
+        set :public_folder, 'public'
+        set :views, 'app/views'
+        enable :sessions
+        set :session_secret, "coffemug"
+    end
+
+    helpers do
+        def logged_in?
+            !session[:id].nil?
+        end
+
+        def current_user
+            session[:id] 
+        end
+    end
+
+
+    get '/' do
+        erb :index
+    end
+
+    get '/signup' do
+
+        if logged_in?
+            redirect '/tweets'  
+        else
+            erb :'/users/create_login'
+        end
+    end
+
+    post '/signup' do
+        if !logged_in?
+            if params[:username] != '' && params[:password] != '' && params[:email] != ''
+                @user = User.create(params)
+                session[:id] = @user.id
+                redirect '/tweets'
+            else
+                #message = all fields must be filled out
+                redirect "/signup"
+            end
+        else
+            # logged in means you can't view sign up page
+            redirect "/tweets"
+        end
+    end
+
+    get '/login' do
+        if logged_in?
+            redirect "/tweets"
+        else
+            erb :"/users/login"
+        end
+    end
+
+    post '/login' do
+
+        if params[:username] != '' && params[:password] != ''
+            user = User.find_by(username: params[:username])
+        end
+        if !!(user && user.authenticate(params[:password]))
+            session[:id] = user.id
+            redirect '/tweets'
+        else
+            redirect '/login'
+        end
+    end
+
+    get "/logout" do
+        session.clear
+        redirect "/login"
+    end
+    post "/tweets/:id/delete" do
+        @tweet = Tweet.find(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.